src/Cache/SourceAndOptionsHashCacheKeyGenerator.php

```diff
@@ -0,0 +1,15 @@
+<?php
+namespace ScriptFUSION\Porter\Cache;
+
+use ScriptFUSION\Porter\Options\EncapsulatedOptions;
+
+class SourceAndOptionsHashCacheKeyGenerator implements CacheKeyGenerator
```

Comment:
[As discussed](https://github.com/ScriptFUSION/Porter/pull/32#discussion_r106402089), this name is meaningless and needs to be renamed to include some strong terms, such as *JSON*, e.g. *JsonCacheKeyGenerator*.